### PR TITLE
[INLONG-4520][Audit] Audit-proxy supports Pulsar authenticate

### LINKED ISSUE #4520: [Improve][Audit] Audit-proxy supports Pulsar authenticate
### Description

Audit-proxy supports Pulsar authenticate

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by doleyzi**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- [INLONG-4520][Audit] Audit-proxy supports Pulsar authenticate

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/incubator-inlong/issues) number)*

- Fixes #4520 

### Modifications

- add pulsar auth switch and token in inlong-audit application.properties

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/pulsar/PulsarClientService.java [hunk 1]
```diff
@@ -76,11 +82,14 @@ public class PulsarClientService {
     public ConcurrentHashMap<String, Producer> producerInfoMap;
     public PulsarClient pulsarClient;
     public String pulsarServerUrl;
+    public boolean pulsarEnableAuth = true;
```
Comment (dockerzhang) [inline inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/pulsar/PulsarClientService.java:85]: pulsarEnableTokenAuth = false

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/pulsar/PulsarClientService.java [hunk 2]
```diff
@@ -76,11 +82,14 @@ public class PulsarClientService {
...
+    public String pulsarAuthToken;
```
Comment (dockerzhang) [inline inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/pulsar/PulsarClientService.java:86]: pulsarTokenAuth

File: inlong-audit/conf/audit-proxy-pulsar.conf [hunk 3]
```diff
@@ -61,6 +61,8 @@ agent1.sinks.pulsar-sink-msg1.channel = ch-msg1
 agent1.sinks.pulsar-sink-msg1.type = org.apache.inlong.audit.sink.PulsarSink
 agent1.sinks.pulsar-sink-msg1.pulsar_server_url = pulsar://localhost:6650
 agent1.sinks.pulsar-sink-msg1.topic = persistent://public/default/inlong-audit
+agent1.sinks.pulsar-sink-msg1.enable_auth = true
```
Comment (dockerzhang) [inline inlong-audit/conf/audit-proxy-pulsar.conf:64]: the default value change to false may be better.

File: inlong-audit/conf/audit-proxy-pulsar.conf [hunk 4]
```diff
@@ -61,6 +61,8 @@ agent1.sinks.pulsar-sink-msg1.channel = ch-msg1
...
+agent1.sinks.pulsar-sink-msg1.auth_token = token
```
Comment (dockerzhang) [inline inlong-audit/conf/audit-proxy-pulsar.conf:65]: the default value change to an empty value.

File: inlong-audit/conf/audit-proxy-pulsar.conf [hunk 5]
```diff
@@ -77,6 +79,8 @@ agent1.sinks.pulsar-sink-msg2.channel = ch-msg2
 agent1.sinks.pulsar-sink-msg2.type = org.apache.inlong.audit.sink.PulsarSink
 agent1.sinks.pulsar-sink-msg2.pulsar_server_url = pulsar://localhost:6650
 agent1.sinks.pulsar-sink-msg2.topic = persistent://public/default/inlong-audit
+agent1.sinks.pulsar-sink-msg1.enable_auth = true
```
Comment (dockerzhang) [inline inlong-audit/conf/audit-proxy-pulsar.conf:82]: ditto

File: inlong-audit/conf/audit-proxy-pulsar.conf [hunk 6]
```diff
@@ -77,6 +79,8 @@ agent1.sinks.pulsar-sink-msg2.channel = ch-msg2
...
+agent1.sinks.pulsar-sink-msg1.auth_token = token
```
Comment (dockerzhang) [inline inlong-audit/conf/audit-proxy-pulsar.conf:83]: ditto

File: inlong-audit/conf/audit-proxy-pulsar.conf (hunk 4 above)
Comment (dockerzhang) [reply inlong-audit/conf/audit-proxy-pulsar.conf:65]: enable_auth
->
enable_token_auth

File: inlong-audit/conf/audit-proxy-pulsar.conf (hunk 6 above)
Comment (doleyzi) [reply inlong-audit/conf/audit-proxy-pulsar.conf]: fix

File: inlong-audit/conf/audit-proxy-pulsar.conf (hunk 4 above)
Comment (doleyzi) [reply inlong-audit/conf/audit-proxy-pulsar.conf]: fix

File: inlong-audit/conf/audit-proxy-pulsar.conf (hunk 3 above)
Comment (doleyzi) [reply inlong-audit/conf/audit-proxy-pulsar.conf]: fix

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/pulsar/PulsarClientService.java (hunk 2 above)
Comment (doleyzi) [reply inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/pulsar/PulsarClientService.java]: fix

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/pulsar/PulsarClientService.java (hunk 1 above)
Comment (doleyzi) [reply inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/pulsar/PulsarClientService.java]: fix

**Review by luchunliang — APPROVED** [review]
LGTM

**Pull request merged**